db/tasks.go

```diff
@@ -176,8 +176,7 @@ func (db *DB) GetTask(id uuid.UUID) (*Task, error) {
 	return r[0], nil
 }
 
-func (db *DB) CreateBackupTask(owner string, job *Job) (*Task, error) {
-	id := uuid.NewRandom()
+func (db *DB) CreateBackupTask(owner string, job *Job, id uuid.UUID) (*Task, error) {
```

Comment:
Maybe we could approach "if uuid is not defined, then generate a new one" ?
Alternative we could split the function into
CreateBackupTaskWithID(...., id )
CreateBackupTas()  as it is